### PR TITLE
Reordering answers

### PR DESCRIPTION
**Pull request by tgandrews**
### What is the context of this PR?
Adds the ability to reorder answers by using the up and down arrows as per the design in survey introduction for collapsibles.

This also bumps the react, react-dom and enzyme versions so we can use hooks! 🎉 

### How to review 
1. Ensure you can re-arrange answers
2. Ensure the animation is smooth
3. Ensure that the re-arrangement is persisted

### CODE REVIEW

**Comment by tgandrews** [comment]
[Re-ordering Answers (6)](https://trello.com/c/dCU00DHG/334-re-ordering-answers-6)

File: eq-author/src/App/questionPage/Design/QuestionPageEditor/AnswersEditor/AnswerEditor/index.js [hunk 1]
```diff
@@ -109,20 +127,52 @@ class AnswerEditor extends React.Component {
...
+          <AnswerType>{this.props.answer.type}</AnswerType>
+
+          <Buttons>
+            <Tooltip
+              content="Move answer up"
+              place="top"
+              offset={{ top: 0, bottom: 10 }}
+            >
+              <MoveButton
+                disabled={!this.props.canMoveUp}
```
Comment (hamishtaplin) [inline eq-author/src/App/questionPage/Design/QuestionPageEditor/AnswersEditor/AnswerEditor/index.js]: As discussed, change this to `aria-disabled` so the button retains focus when "disabled".

File: eq-author/src/App/questionPage/Design/QuestionPageEditor/AnswersEditor/AnswerEditor/MoveButton.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  &:hover {
+    color: ${colors.black};
+  }
+
+  &:focus {
+    outline: 3px solid ${colors.orange};
+  }
+
+  &[disabled] {
```
Comment (hamishtaplin) [inline eq-author/src/App/questionPage/Design/QuestionPageEditor/AnswersEditor/AnswerEditor/MoveButton.js]: Style the icon as disabled, so the focus style is still full opacity:

```css
&[aria-disabled=true] {
  svg {
    opacity: 0.3;
  }
}
```

File: eq-author/src/App/questionPage/Design/QuestionPageEditor/AnswersEditor/AnswerEditor/index.js [hunk 3]
```diff
@@ -23,39 +24,56 @@ import {
...
+  padding: 1em 6em 1em 1.5em;
 `;
 
-export const AnswerDeleteButton = styled(DeleteButton)`
-  position: absolute;
-  right: 0.2em;
-  top: 1em;
+const Buttons = styled.div`
+  display: flex;
+  padding-right: 0.3em;
```
Comment (hamishtaplin) [inline eq-author/src/App/questionPage/Design/QuestionPageEditor/AnswersEditor/AnswerEditor/index.js]: I regret adding this padding now:

```suggestion
```

**Pull request merged**